internal/sirius/get_user_details_test.go

```diff
@@ -0,0 +1,83 @@
+package sirius
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/pact-foundation/pact-go/dsl"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetUserDetails(t *testing.T) {
+	t.Parallel()
+
+	pact := newPact()
+	defer pact.Teardown()
+
+	testCases := []struct {
+		name             string
+		setup            func()
+		expectedResponse User
+		expectedError    func(int) error
+	}{
+		{
+			name: "OK",
+			setup: func() {
+				pact.
+					AddInteraction().
+					Given("A donor exists").
+					UponReceiving("A request for the current user").
+					WithRequest(dsl.Request{
+						Method: http.MethodGet,
+						Path:   dsl.String("/lpa-api/v1/users/current"),
+					}).
+					WillRespondWith(dsl.Response{
+						Status: http.StatusOK,
+						Body: dsl.Like(map[string]interface{}{
+							"id":          dsl.Like(103),
+							"displayName": dsl.String("Test User"),
+							"roles":       dsl.Like([]string{"OPG User", "Case Manager", "Reduced Fees User"}),
```

Comment:
I wanted to add a cypress test that also checks that the apply fee reduction button is not shown for users without the reduced fees user role but as i can tell i can't add another mock for this same endpoint..

Reply:
Arg yeah thats a pain...